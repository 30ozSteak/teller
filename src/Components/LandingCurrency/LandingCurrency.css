.currency-card {
  background: linear-gradient(to right, #424b6e, #272e4e);
  border-radius: 15px;
  box-shadow: 5px 5px 15px rgba(0, 0, 0, 0.459);
  color: var(--AlmostWhite);
  margin: 0.5rem 0;
  transition: 0.7s;
  width: 95%;
}

.card-inside-text {
  align-items: center;
  display: grid;
  grid-template-columns: 50px 3rem 7rem 5rem 30px;
  height: 5rem;
  padding: 0 1rem;
}

.currency-card:hover {
  filter: brightness(1.3);
}

.expanded-currency {
  display: flex;
  flex-wrap: wrap;
  padding: 0.5rem 0;
  justify-content: space-around;
  text-transform: uppercase;
  transition: 0.2s;
  z-index: 90;
  position: relative;
  top: 0;
}

.expanded-graph {
  transition: 0.2s;
}

.cc-left img {
  background: var(--Peachy);
  border-radius: 5px;
  height: 20px;
  padding: 5px;
  width: 20px;
}

.cc-center p:first-child {
  font-size: 13px;
  font-weight: 500;
}

.cc-center-right {
  font-size: 1rem;
  text-align: center;
}

.cc-center p:nth-child(2) {
  color: rgba(255, 255, 255, 0.609);
  font-size: 12px;
}

.ec-right {
  background: linear-gradient(to right, #424b6e, #272e4e);
  border-radius: 10px;
  color: var(--AlmostWhite);
  height: 2em;
  margin-bottom: 10px;
  width: 4em;
}

.ec-right:hover {
  cursor: pointer;
}

.fave-this {
  fill: #000;
  height: 20px;
  position: relative;
  right: 0rem;
  transition: 0.2s;
  z-index: 50;
}

.fave-this:hover {
  cursor: pointer;
}

.cc-right {
  text-align: center;
}

.expansion-dots-glowing,
.expansion-dots {
  cursor: pointer;
  height: 100%;
}

.expansion-dots div,
.expansion-dots-glowing div {
  background-color: rgba(109, 109, 109, 0.514);
  border-radius: 50%;
  height: 5px;
  margin: 2px 0px 2px 10px;
  position: relative;
  width: 5px;
}

.expansion-dots-glowing div:first-child,
.expansion-dots div:first-child {
  margin-top: 2rem;
}

.expansion-dots-glowing div {
  animation: wavey 2s linear infinite;
}

.expansion-dots-glowing div:first-child {
  animation-delay: 0.5s;
}

.expansion-dots-glowing div:nth-child(2) {
  animation-delay: 1s;
}

.expansion-dots-glowing div:nth-child(3) {
  animation-delay: 1.5s;
}

@keyframes wavey {
  0% {
    filter: brightness(1);
  }
  50% {
    filter: brightness(3.5);
  }
  100% {
    filter: brightness(1);
  }
}

@media only screen and (min-width: 700px) {
  .currency-card {
    width: 99%;
  }
  .currency-container {
    width: 25rem;
  }
}
