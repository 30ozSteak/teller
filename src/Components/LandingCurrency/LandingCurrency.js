import React, { Component } from "react";
import "./LandingCurrency.css";
import Heart from "../../Assets/heart.svg";
import HeartP from "../../Assets/heartpink.svg";
import { Icons } from "../../Assets/cryptoIcons/cryptoIcons";
import DataCleaner from "../../Utils/Cleaners/";
import { sendFavorites } from "../../Utils/API/";
// import LineChart from "../LineChart/LineChart"
class LandingCurrency extends Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false,
      faved: false,
      token: this.props.token
    };
    this.cleaner = new DataCleaner();
  }

  expand = () => {
    this.setState({
      expanded: !this.state.expanded
    });
  };

  faved = () => {
    this.setState({
      faved: !this.state.faved
    });
  };

  handleClick = async e => {
    const { name } = e.target;
    this.props.displayExpanded(name)
  };

  handleFaveClick = async e => {
    // const { name } = e.target;
    const { addToFavorites, currency } = this.props
    sendFavorites(JSON.stringify({
      "name": currency.name,
      "price_usd": currency.price,
      "percent_change_24_hr": currency.percent_change
    }), this.props.token)
    addToFavorites({
      name: currency.name,
      price_usd: currency.price,
      percent_change: currency.percent_change
    });
    this.faved();
  }

  render() {
    const { symbol, name, price, percent_change } = this.props.currency;
    const { expanded } = this.state;
    return (
      <div className="currency-card">
        <div className="card-inside-text">
          <div className="cc-left">
            <img src={Icons[this.props.currency.symbol]} alt="" />
          </div>
          <div className="cc-center">
            <p>{symbol}</p>
            <p>{name}</p>
          </div>
          <div className="cc-center-right">
            <p>$ {price}</p>
          </div>
          <div className="cc-right">
            <p>{percent_change}</p>
          </div>
          <div
            className={expanded ? "expansion-dots-glowing" : "expansion-dots"}
            onClick={this.expand}
          >
            <div />
            <div />
            <div />
          </div>
        </div>
        {this.state.expanded && (
          <div className="expanded-currency">
            <button
              className="ec-right"
              onClick={this.handleClick}
              name={name}>
              View
            </button>
            <img
              name={name}
              src={this.state.faved ? HeartP : Heart}
              className="fave-this"
              onClick={this.handleFaveClick}
              alt=""
            />
          </div>
        )}
        {this.state.expanded && (
          <div className="expanded-graph">
            {/*<LineChart />*/}
          </div>
        )}
      </div>
    );
  }
}

export default LandingCurrency;
