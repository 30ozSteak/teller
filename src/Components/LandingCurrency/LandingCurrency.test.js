import React from 'react'
import LandingCurrency from './LandingCurrency'
import { shallow } from 'enzyme'

describe('LandingCurrency Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<LandingCurrency />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})