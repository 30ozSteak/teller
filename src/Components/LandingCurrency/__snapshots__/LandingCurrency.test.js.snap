// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingCurrency Component should match the snapshot 1`] = `undefined`;
